### PR TITLE
Drop support for GCC on OS X :(

### PR DESCRIPTION
**Pull request by mrks**
https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/749200
We have to drop support for GCC on OS X because it caused segfaults when used with tbb.

When I find the time, I will see if I can fix the symbols exported by TBB, but that might take some time.

### CODE REVIEW

**Comment by TwoLaid** [comment]
😢

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,12 +7,21 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.2)
         message(FATAL_ERROR "Your GCC version ${CMAKE_CXX_COMPILER_VERSION} is too old.")
     endif()
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+    if (APPLE)
+        # https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/749200
+        message(WARNING "We had to drop support for GCC on OS X because it caused segfaults when used with tbb. You can continue, but don't hold us responsible for any segmentation faults.")
+        execute_process(COMMAND read -p "Press [enter] to continue anyway")
```
Comment (mrzzzrm) [inline CMakeLists.txt]: Wouldn't consider this the proper way to query the user for input. In CMake you expose options (ENABLE_OSX_GCC_SUPPORT defaulting to FALSE, ...) and check for them. Don't know how IDEs would handle this (probably stall...) but its unclean anyway, so let's use options.

https://cmake.org/cmake/help/v3.9/command/option.html

Comment (mrks) [reply CMakeLists.txt]: I tend to forget that "IDEs" exist. ;)

Will fix as proposed, but probably not before the holidays.

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,12 +7,21 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
...
+    endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
```
Comment (mjendruk) [inline CMakeLists.txt]: While we’re at it, we should also update the minimum required Clang version. `4.0` does not support all C++17 feature we currently use (e.g. “Template argument deduction for class templates”). It should be `5.0`.

Comment (mrks) [reply CMakeLists.txt]: done

File: CMakeLists.txt (hunk 1 above)
Comment (mrks) [reply CMakeLists.txt]: done

**Comment by mrks** [comment]
Please re-review and test on your linux machine.

**Comment by Bensk1** [comment]
There was an update for TBB on macOS, does this change anything?

**Comment by mrks** [comment]
I scrolled over the [commit](https://github.com/01org/tbb/commit/4c73c3b5d7f78c40f69e0c04fd4afb9f48add1e6) and did not find anything that would lead me to believe that the problem has been fixed.

**Pull request merged**